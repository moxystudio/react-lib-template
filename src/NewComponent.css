.container {
    background-color: red;

    &::after {
        background-color: pink;
    }
}
