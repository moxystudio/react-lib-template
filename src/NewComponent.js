// Here, create the new Component you want to see exported to a new package
import React from 'react';
import PropTypes from 'prop-types';

const NewComponent = ({ children }) => (
    <div>
        { children }
    </div>
);

NewComponent.propTypes = {
    children: PropTypes.any.isRequired,
};

export default NewComponent;
