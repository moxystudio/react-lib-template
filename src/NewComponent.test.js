import React from 'react';
import { render } from '@testing-library/react';
import NewComponent from './NewComponent';

const defaultProps = {
    children: <span>Hello</span>,
};

const renderWithProps = (props = {}) => render(<NewComponent { ...defaultProps } { ...props } />);

it('should render correctly', () => {
    const { getByText } = renderWithProps();

    expect(getByText('Hello')).toBeInTheDocument();
});

it('should render correctly with different children', () => {
    const { asFragment } = renderWithProps({
        children: <span>Foo Bar</span>,
    });

    expect(asFragment()).toMatchSnapshot();
});
