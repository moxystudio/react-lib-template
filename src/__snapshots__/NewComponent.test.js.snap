// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with different children 1`] = `
<DocumentFragment>
  <div
    class="next-lib-template_container"
  >
    <span>
      Foo Bar
    </span>
  </div>
</DocumentFragment>
`;
